src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/IHtmlHelper.cs

```diff
@@ -169,7 +169,12 @@ public interface IHtmlHelper
 
         /// <summary>
         /// Returns an &lt;input&gt; element of type "checkbox" with value "true" and an &lt;input&gt; element of type
-        /// "hidden" with value "false".
+        /// "hidden" with value "false" for the specified <paramref name="expression"/>. Adds a "checked" attribute to
+        /// the "checkbox" element if first non-<c>null</c> value found is <c>true</c>. Looks for
```

Comment:
1. Not sure I understand the "if first" thing? Can we simplify that to cover the actual common case?
2. The stuff about the "checked" attribute sounds a bit nerdy to me anyway. I was thinking something more like, "The checkbox will be set as checked if X, Y, or Z."